src/lib.rs

```diff
@@ -1316,6 +1444,13 @@ impl Bump {
     fn alloc_layout_slow(&self, layout: Layout) -> Option<NonNull<u8>> {
         unsafe {
             let size = layout.size();
+            let allocation_limit_remaining = self.allocation_limit_remaining();
+
+            // This is also checked within `new_chunk` but the check here
+            // acts as an early exit if the request could never be satisfied given the limit
+            if !Bump::new_allocation_fits_under_limit(allocation_limit_remaining, size) {
+                return None;
+            }
```

Comment:
We should only check this once, so either

* let's remove the check here, since `Bump::new_chunk` will check it, or
* let's remove the check in `Bump::new_chunk`, and only check it here.

I'm partial to the second option, since I think we won't need the extra parameter for `Bump::new_chunk` anymore. Although the check will have to move down into the `sizes` iterator below, to account for the size of the chunk, rather than just the size of the object (which is typically much smaller than the size of the chunk).